### PR TITLE
[amp-stories sub-task] Split Grid Layer Block

### LINKED ISSUE #1526: Blocks > Do not allow more than one block inside the AMP Grid layer when the 'fill' template is selected
**As a editor/author creating an AMP story I want to be prevented from adding more than one block to the AMP Grid Layer when using the 'fill template' so that I don't hide all the 'other' blocks by mistake.** 

- [ ] _AC 1:_  When the fill template is selected for the AMP Grid Layer, only 1 block can be added to the Grid Layer
- [ ] _AC2:_ Appropriate messaging in the interface will let the user know that when using the 'fill template' on the AMP Grid Layer, only 1 block can be added to the AMP Grid layer
- [ ] _AC3:_ If the user tries to add a 2nd block to a AMP Grid layer with the template set to fill, a warning will display for the user and the user will not able able to add the 2nd block
- [ ] _AC4:_ For other templates (i.e. the vertical one) multiple blocks can be added to an AMP Grid layer, and blocks can be laid out in relation to each other

_Background on this change_ 
This brings the functionality in alignment with the expected functionality on the FE and avoids user confusion when the template fill seems to ‘hide’ all other blocks except the first one.  From a user perspective the AMP Grid Layer with Template ‘Fill’ hides all the other blocks except the first one. This can cause issues using the  default paragraph block in a layer when background image fill is set.  ie. see this example where the image block added first covers/hides the 'text' block.   https://d.pr/i/gHSAxO

### LINKED ISSUE #1530: Blocks > Make AMP Stories block more granular by providing different blocks for different layout options
**As AMP story creator, I'd like the AMP Stories block to be more granular so I can split the grid layer, for different layout options.** 
 - [ ] _AC 1:_ Split up the grid layer into separate blocks: vertical grid layer, thirds grid layer, horizontal grid layer, fill grid layer. 
 - [ ] _AC 2:_ The fill grid layer will be able to be further specialized with more preconfigured block configurations for background fill image layer and background fill video layer (NB: this will lay the foundation for user being able to save new block templates which are pre-populated with the common configurations of blocks that they use) 
- [ ] _AC 3:_ Provide better defaults for blocks in AMP Stories, in particular the video block ie.: 
     - [ ] Template is set to “fill”
     - [ ] AMP layout default value is set to “fill”
     - [ ] Autoplay (block option for autoplay/not auto-play

### PR DESCRIPTION
**Pull request by miina**
Removes the select for `template` and splits the AMP Story Grid Layer block into the following blocks:
- Horizontal Layer;
- Vertical Layer;
- Thirds Layer;
- Background Image Layer;
- Background Video Layer;

Background Image and Background Video Layer replace the previous Fill template. Both of these allow only one child block -- Image / Video respectively and are already pre-populated with the block.

Fixes #1530;
Fixes #1526;

### CODE REVIEW

**Comment by miina** [comment]
@jacobschweitzer Would you mind reviewing the PR? It's still missing the correct SVG-s for each block but has placeholders for now -- everything else should be ready, will replace the SVG-s as soon as I get them.

**Comment by cathibosco** [comment]
@miina I outlined all of the paths, I separated them out into individual files, I am including an optimized version as well and kept with a consistent naming system.
Please let me know if these will work well or if you need any changes. Thank you
[Archive.zip](https://github.com/Automattic/amp-wp/files/2565946/Archive.zip)

File: assets/css/amp-editor-story-blocks.css [hunk 1]
```diff
@@ -137,28 +137,28 @@ div[data-amp-position="lower-third"] {
...
 	display: none;
 }
 
-.editor-block-list__layout div[data-type="amp/amp-story-grid-layer"].is-selected ~ div[data-type="amp/amp-story-grid-layer"],
-.editor-block-list__layout div[data-type="amp/amp-story-grid-layer"].is-selected-parent ~ div[data-type="amp/amp-story-grid-layer"],
-.editor-block-list__layout div[data-type="amp/amp-story-grid-layer"].is-selected ~ div[data-type="amp/amp-story-cta-layer"],
-.editor-block-list__layout div[data-type="amp/amp-story-grid-layer"].is-selected-parent ~ div[data-type="amp/amp-story-cta-layer"]{
+.editor-block-list__layout div[data-amp-type="grid"].is-selected ~ div[data-amp-type="grid"],
+.editor-block-list__layout div[data-amp-type="grid"].is-selected-parent ~ div[data-amp-type="grid"],
+.editor-block-list__layout div[data-amp-type="grid"].is-selected ~ div[data-amp-type="grid"],
```
[the remark this answers is not in the document]
Comment (miina) [reply assets/css/amp-editor-story-blocks.css]: Correct, thanks for finding this! Will change.

**Comment by cathibosco** [comment]
Revised SVG icons - I think I am cross posting this - sorry if it causes any confusion... 
Version2

**Comment by cathibosco** [comment]
Version 3 Goldilocks: 
[Archive-v3.zip](https://github.com/Automattic/amp-wp/files/2567026/Archive-v3.zip)

**Pull request merged**